x-pack/plugins/ml/public/application/model_management/add_model_flyout.tsx

```diff
@@ -0,0 +1,444 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  EuiBadge,
+  EuiButton,
+  EuiButtonEmpty,
+  EuiCheckableCard,
+  EuiCodeBlock,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFlyout,
+  EuiFlyoutBody,
+  EuiFlyoutFooter,
+  EuiFlyoutHeader,
+  EuiFormFieldset,
+  EuiIcon,
+  EuiLink,
+  EuiSpacer,
+  EuiSteps,
+  EuiTab,
+  EuiTabs,
+  EuiText,
+  EuiTitle,
+  EuiToolTip,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+import React, { type FC, useMemo, useState } from 'react';
+import { groupBy } from 'lodash';
+import { usePermissionCheck } from '../capabilities/check_capabilities';
+import { useMlKibana } from '../contexts/kibana';
+import { ModelItem } from './models_list';
+
+export interface AddModelFlyoutProps {
+  modelDownloads: ModelItem[];
+  onClose: () => void;
+  onSubmit: (modelId: string) => void;
+}
+
+export const AddModelFlyout: FC<AddModelFlyoutProps> = ({ onClose, onSubmit, modelDownloads }) => {
+  const {
+    services: { docLinks },
+  } = useMlKibana();
+
+  const [selectedTabId, setSelectedTabId] = useState('elser');
+  const [selectedModelId, setSelectedModelId] = useState<string | undefined>(
+    modelDownloads.find((m) => m.recommended)?.model_id
+  );
+
+  const canCreateTrainedModels = usePermissionCheck('canCreateTrainedModels');
+
+  const tabs = useMemo(() => {
+    return [
+      ...(canCreateTrainedModels
```

Comment:
I think you should add an extra check here to also hide the ELSER tab if both versions of the model are already downloaded:
<img width="353" alt="image" src="https://github.com/elastic/kibana/assets/7405507/a9d1dc1c-b0a8-420f-ace7-bfb74917904b">

Reply:
Added in https://github.com/elastic/kibana/pull/171024/commits/ebde32935b8ab8499c0531409e37a96c207e624a